### ISSUE TITLE
Use `WAL` as default everywhere

### ISSUE THREAD

**Issue by avinassh** [tursodatabase/libsql#1553]
Currently, Turso (the platform), `libsql-sever`, and embedded replicas use WAL mode and it is the only mode supported. However `libsql` drivers don't have WAL mode as default. Let's change that to use WAL everywhere. 

reported here: https://discord.com/channels/933071162680958986/1154725980183732224/1261642078694150144

**Comment by imskr** [comment]
Can I work on this? @avinassh please assign me

**Comment by avinassh** [comment]
@imskr sure! feel free to work on it

I would start with the Rust driver first: https://github.com/tursodatabase/libsql/tree/main/libsql

**Comment by imskr** [comment]
Thanks @avinassh I was looking to set 

```rs
PRAGMA journal_mode = wal
```

In the rust driver I don't see journal_mode, Can you point me to the right place?

**Comment by imskr** [comment]
Any pointer on this @avinassh ?

**Comment by avinassh** [comment]
I don't think it is that simple as we don't set any journal mode and use the defaults. I would probably find the methods where we are creating new connections and setup the journal mode appropriately.
